### PR TITLE
Should use viz.js when graphviz is not installed

### LINKED ISSUE #25: Should use viz.js when graphviz is not installed
A lot of npm users would benefit by just doing an `npm install node-plantuml` without having to worry having graphviz installed on their system. Currently, there is a way to do this by using viz.js, which is graphviz compiled using emscripten, and then wrapping that inside a jar by executing with V8. This sounds convoluted but it works well. See http://plantuml.com/vizjs

### PR DESCRIPTION
**Pull request by atifsyedali**
Resolves #25

### CODE REVIEW

**Comment by markushedvall** [comment]
Prepublish is used to ensure installers are not required to have access to sourceforge, see https://github.com/markushedvall/node-plantuml/issues/20

Either we need to publish with the jar for each platform, which doesn't sound great, or keep the get-plantuml-jar.js unchanged and add a seperate get-vizjs.js that can run during postinstall.

I think adding a separate script for retrieving the vizjs jar files is the way to go.

File: scripts/get-plantuml-jar.js [hunk 1]
```diff
@@ -29,19 +38,58 @@ function download (uri, filename) {
...
     }
   })
 }
 
 console.log('Downloading plantuml.jar version ' + plantumlVersion)
-download(PLANTUML_FILES_URL + PLANTUML_FILES_JAR_PATH, PLANTUML_JAR)
+download(PLANTUML_FILES_URL + PLANTUML_FILES_JAR_PATH, PLANTUML_JAR, function (err) {
+  if (!err) {
+    // once downloaded, check if dot is installed using plantuml's testdot command
+    var plantuml = require('../lib/node-plantuml')
```
Comment (markushedvall) [inline scripts/get-plantuml-jar.js]: Could you explain how we know node-plantuml is always available in this location?

Comment (atifsyedali) [reply scripts/get-plantuml-jar.js]: Given `require(x)`, where x is a relative path, x will always be relative to the directory of the containing module.

**Comment by atifsyedali** [comment]
> Prepublish is used to ensure installers are not required to have access to sourceforge, see #20
> 
> Either we need to publish with the jar for each platform, which doesn't sound great, or keep the get-plantuml-jar.js unchanged and add a seperate get-vizjs.js that can run during postinstall.
> 
> I think adding a separate script for retrieving the vizjs jar files is the way to go.

Interesting, didn't know about #20. I like the separate get-vizjs.js during postinstall. Then it is up to the dependants to ensure that the build system has graphviz installed. However I will make sure that the postinstall script doesn't exit with an error code -- just an error is output in console.

**Comment by atifsyedali** [comment]
Reverted the change for getting plantuml.jar from postinstall back to prepublish. Also, downloading vizjs jars is now a separate postinstall script that does not exit with an error if the download is blocked (error is still printed in console).

File: scripts/get-plantuml-jar.js (hunk 1 above)
Comment (markushedvall) [reply scripts/get-plantuml-jar.js]: Yes, of course. Not sure was I thinking :)

**Comment by markushedvall** [comment]
So far I've only tested it on Linux but I've had some mixed result.

The installation work as expected so the problems I'm experiencing is with plantuml using vizjs. Basically every time I try to generate a diagram an exception is thrown in the java process (V8ScriptExecutionException). But most of the time I still get a diagram that looks fine. Sometimes I get a diagram that contain the whole stack trace instead.

I'm still fine with adding this as an experimental feature.

A few notes:
- Installing without graphviz on the system and then installing graphviz afterwards will result in real dot being used.
- vizjs can still be forced to be used by setting env. variable GRAPHVIZ_DOT=vizjs
- It might be a good idea to let someone who have grahpviz on the system be able to still download the vizjs and v8 jars anyway by setting some env. variable.

**Pull request merged**